platforms/r3-corda-ent/configuration/roles/setup/credentials/tasks/cenm_tasks.yaml

```diff
@@ -57,12 +57,31 @@
 # Write the ssl credentials to the vault
 - name: Write the ssl credentials to the vault if they dont exist
   shell: |
-    vault kv put {{ org.vault.secret_path | default('secretsv2') }}/{{ org.name | lower }}/credentials/ssl idman={{ org.credentials.ssl.idman }} networkmap={{ org.credentials.ssl.networkmap }} signer={{ org.credentials.ssl.signer }} root={{ org.credentials.ssl.root }}
+    vault kv put {{ org.vault.secret_path | default('secretsv2') }}/{{ org.name | lower }}/credentials/ssl idman={{ org.credentials.ssl.idman }} networkmap={{ org.credentials.ssl.networkmap }} signer={{ org.credentials.ssl.signer }} root={{ org.credentials.ssl.root }} auth={{ org.credentials.ssl.auth }}
   environment:
     VAULT_ADDR: "{{ org.vault.url }}"
     VAULT_TOKEN: "{{ org.vault.root_token }}"
   when: vault_credentials_ssl.failed == True
 
+# Check if the initial user credentials are already present in the vault
+- name: Check if the initial user credentials are already present in the vault
+  shell: |
+    vault kv get -field=user {{ org.vault.secret_path | default('secretsv2') }}/{{ org.name | lower }}/credentials/user
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: vault_credentials_user
+  ignore_errors: yes
+
+# Write the initial user credentials to the vault
+- name: Write the initial user credentials to the vault if they dont exist
+  shell: |
+    vault kv put {{ org.vault.secret_path | default('secretsv2') }}/{{ org.name | lower }}/credentials/user username=admin userpwd=p4ssWord
```

Comment:
Shouldnt this username password been taken from network.yaml as well?

Reply:
fetched username and password from network.yaml